AppCasasAPI/AppCasasAPI/Repository/ReferenceDataRepository.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Npgsql;
+using AppCasasAPI.Repository.Interfaces;
+using AppCasasAPI.Dto.Response;
+using AppCasasAPI.Model;
+
+namespace AppCasasAPI.Repository
+{
+    public class ReferenceDataRepository : IReferenceDataRepository
+    {
+        private readonly NpgsqlDataSource _dataSource;
+
+        public ReferenceDataRepository(NpgsqlDataSource dataSource) {
+            _dataSource = dataSource;
+        }
+
+        public async Task<ReferenceDataResponseDto> GetAllReferenceDataAsync()
+        {
+            ReferenceDataResponseDto refData = new();
+            using (var conn = await _dataSource.OpenConnectionAsync())
+            {
+                using var typologyQuerry = new NpgsqlCommand("SELECT * FROM tipologia;", conn);
+                var typologyReader = await typologyQuerry.ExecuteReaderAsync();
```

Comment:
When we ExecuteReaderAync() i believe we can use an "using" statement and delete the reader.Close().
can you try it and let me know if it works?

Also consider using a try/catch blocks to prevent exceptions, inside the executeReader using.